tests/testthat/test-paginate_listing.R

```diff
@@ -111,29 +110,43 @@ testthat::test_that("horizontal pagination with 0 or 1 key column specified work
 testthat::test_that("listing works with no vertical pagination", {
   # pre-processing and ordering
   tmp_data <- ex_adae %>%
-    dplyr::select(USUBJID, AGE, BMRKR1) %>%
     dplyr::slice(1:30) %>%
-    distinct()
+    dplyr::distinct(USUBJID, AGE, BMRKR1, .keep_all = TRUE)
 
   lsting <- as_listing(tmp_data,
-                       key_cols = c("USUBJID", "AGE")
+                       key_cols = c("USUBJID", "AGE"),
+                       disp_cols = character()
   ) %>%
     add_listing_col("BMRKR1", format = "xx.x")
 
-  pages_listings <- paginate_listing(lsting, lpp = NULL)
+  pages_listings <- paginate_listing(lsting, lpp = NULL, verbose = TRUE)
+  page1_result <- matrix_form(pages_listings[[1]])
 
+  testthat::expect_equal(length(pages_listings), 1)
+  testthat::expect_equal(attr(page1_result, "ncol"), 3)
```

Comment:
Why are you using a raw `attr` call here? we want to try very hard to avoid that as it's brittle